lib/bundler/dsl.rb

```diff
@@ -449,6 +451,10 @@ def warn_deprecated_git_source(name, repo_string)
       EOS
     end
 
+    def generate_checksum(gemfile_source)
+      Digest::MD5.hexdigest(gemfile_source)
```

Comment:
We can't use MD5 because of FIPS :( https://github.com/bundler/bundler/issues/4989